lib/streamlit/runtime/app_session.py

```diff
@@ -474,6 +475,7 @@ def _on_scriptrunner_event(
         client_state: ClientState | None = None,
         page_script_hash: str | None = None,
         fragment_ids_this_run: set[str] | None = None,
+        pages: dict[PageHash, PageInfo] = {},
```

Comment:
Same comment about the default value used here

Reply:
I can probably address the others. This one probably cannot cause of the way the event structure is worked. There's got to be a default for each one otherwise, every script runner event requires it (which seems over the top).

It perhaps suggests that having one event handler for any event going on in Streamlit might not be the most scalable.